ceph/rados/mgr_workflows.py

```diff
@@ -0,0 +1,294 @@
+            )
+
+        if mgr_exists:
+            log.error(
+                f"Mgr daemon still present on the host {host_name}. Failed to remove"
+            )
+            return False
+
+        # Removing mgr label from host if present
+        labels = self.mon_obj.get_host_labels(host=host_name)
+        if "mgr" in labels:
+            cmd = f"ceph orch host label rm {host_name} mgr"
+            self.client.exec_command(sudo=True, cmd=cmd)
+            log.debug(f"Removed mgr label from host : {host_name}")
+            time.sleep(30)
+
+        # Unsetting mgr daemon
+        managed_chk = self.rados_obj.set_managed_flag("mgr")
+        time.sleep(30)
+        if not managed_chk:
+            log.error("The unmanaged flag is  not unset")
+            return False
+
+        log.debug(f"Mgr successfully removed from host : {host_name}")
+        return True
+
+    def mgr_scale_up_down(self, mgr_node_list):
+        """
+        Method to scale up or down the mgr daemons to the cluster
+        Args:
+            mgr_node_list : This parameter contain the mgr daemon list.If mgr_node_list is greater than the
+            current mgr list then the method performs scale up.If its less the method performs scale down.
+        returns:
+            Pass -> True, Fail -> false
+        """
+        mgr_daemons = self.get_mgr_daemon_list()
+        nodes_list = [item.split(".")[0] for item in mgr_daemons]
+        log.info(f"The mgr_daemons list is-{mgr_daemons}")
+        log.info(f"The new mgr list is -{mgr_node_list}")
+        mgr_count = len(mgr_daemons)
+        new_mgr_count = len(mgr_node_list)
+
+        additional_mgr_count = new_mgr_count - mgr_count
+        if mgr_count == new_mgr_count:
+            log.info(" The request count is same as the current mgr count")
+            if set(nodes_list) == set(mgr_node_list):
+                log.info("The new mgr nodes list is same as old list")
+            return True
+
+        new_mgr = []
+        if new_mgr_count > mgr_count:
+            log.info(
+                f"Performing MGR scale up.The mgr is scale-up from {mgr_count} to {new_mgr_count} "
+            )
+            scale_up = True
+            # Node list to add the label
+            new_mgr = mgr_node_list[-additional_mgr_count:]
+        else:
+            log.info(
+                f"Performing MGR scale down.The mgr is scale-down from {mgr_count} to { new_mgr_count} "
+            )
+            # Node list to remove the label
+            new_mgr_daemon = [
+                item
+                for item in mgr_daemons
+                if not item.startswith(tuple(mgr_node_list))
+            ]
+            for item in new_mgr_daemon:
+                new_mgr.append(item.split(".")[0])
+            scale_up = False
+
+        mgr_nodes = "  ".join(mgr_node_list)
+        cmd = f'ceph orch apply mgr "{new_mgr_count}  {mgr_nodes}"'
```

Comment:
To increase and decrease, we can just remove and add labels since the deployment placement spec is already based on labels.

Here we are again changing the count of the no of MGR's, so the placement is being changed here from "label:mgr" to "count:<count>"

Should be fine for now, but please create a backlog task to fix this.

Reply:
Created the following backlog task to update the code.
https://issues.redhat.com/browse/RHCEPHQE-14837